crystal/src/y3storage/autoinst_issues/invalid_value.cr

```diff
@@ -0,0 +1,103 @@
+# Copyright (c) [2019-2020] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "./issue"
+
+module Y3Storage
+  module AutoinstIssues
+    # Represents an AutoYaST situation where an invalid value was given.
+    #
+    # Example: invalid value 'auto' for attribute :size on /home partition
+    #     section = AutoinstProfile::PartitioningSection.new_from_hashes({"size" => "auto"})
+    #     problem = InvalidValue.new(section, :size, section.size)
+    #     problem.value #=> "auto"
+    #     problem.attr  #=> :size
+    class InvalidValue < Issue
+      # Section where it was detected (see {AutoinstProfile})
+      getter section : AutoinstProfile::PartitionSection | AutoinstProfile::SkipListSection
+
+      # Name of the missing attribute
+      getter attr : Symbol
+
+      # Invalid value
+      getter value : Int64 | String
+
+      # New value or :skip to skip the section.
+      getter new_value : Int64 | String | Symbol
```

Comment:
can crystal specify e.g. `:skip` instead of generic symbol? So it catches during compilation that you use e.g. wrong symbol like `:skyp`

Reply:
No, as far as I know.